dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java

```diff
@@ -210,6 +210,7 @@
     E4310( "SQL query contains references to protected tables" ),
     E4311( "SQL query contains illegal keywords" ),
     E4312( "Current user is not authorised to read data from SQL view: `{0}`" ),
+    E4313( "SQL query contains variable names that are invalid: `{0}`" ),
```

Comment:
Would this be a bit more accurate?

"SQL query variable contains invalid characters: `{0}`"

Reply:
No I think that would be confusing. The query itself does not really have invalid characters. Just that the placeholder or variable name which will be replaced is not named according to what we allow.